python/hail/docs/hailpedia/matrixtable.rst

```diff
@@ -0,0 +1,404 @@
+-----------
+MatrixTable
+-----------
+
+A :class:`.MatrixTable` is a distributed two-dimensional dataset consisting of
+four components: a two-dimensional matrix where each entry is indexed by row
+key(s) and column key(s), a corresponding rows table that stores all of the row
+fields which are constant for every column in the dataset, a corresponding
+columns table that stores all of the column fields that are constant for every
+row in the dataset, and a set of global fields that are constant for every entry
+in the dataset.
+
+Unlike a :class:`.Table` which has two field groups (row fields and global
+fields), a matrix table has four field groups: global fields, row fields, column
+fields, entry fields.
+
+In addition, there are different operations on the matrix for each field group.
+For instance, :class:`.Table` has :meth:`.Table.select` and
+:meth:`.Table.select_globals`, and :class:`.MatrixTable` has
+:meth:`.MatrixTable.select_rows`, :meth:`.MatrixTable.select_cols`,
+:meth:`.MatrixTable.select_entries`, and :meth:`.MatrixTable.select_globals`.
+
+It is possible to represent matrix data by coordinate in a table , storing one
+record per entry of the matrix. However, the :class:`.MatrixTable` represents
+this data far more efficiently and exposes natural interfaces for computing on
+it.
+
+The :meth:`.MatrixTable.rows` and :meth:`.MatrixTable.cols` methods return the
+row and column fields as separate tables. The :meth:`.MatrixTable.entries`
+method returns the matrix as a table in coordinate form -- use this object with
+caution.
+
+Keys
+====
+
+Matrix tables have keys just as tables do. However, instead of one key, matrix
+tables have two keys: a row key and a column key. Row fields are indexed by the
+row key, column fields are indexed by the column key, and entry fields are
+indexed by the row key and the column key. The key structs can be accessed with
+:attr:`.MatrixTable.row_key` and :attr:`.MatrixTable.col_key`. It is possible to
+change the key with :meth:`.MatrixTable.key_rows_by` and
+:meth:`.MatrixTable.key_cols_by`.
+
+Note that changing the row key, however, may be an expensive operation.
+
+Hail matrix tables are natively distributed objects, and as such have another
+key: a partition key. This key is used for specifying the ordering of the matrix
+table along the row dimension, which is important for performance. Access this
+with :attr:`.MatrixTable.partition_key`
+
+Referencing Fields
+==================
+
+All fields (row, column, global, entry) are top-level and exposed as attributes
+on the :class:`.MatrixTable` object. For example, if the matrix table `mt` had a
+row field `locus`, this field could be referenced with either ``mt.locus`` or
+``mt['locus']``. The former access pattern does not work with field names with
+spaces or punctuation.
+
+The result of referencing a field from a matrix table is an :class:`.Expression`
+which knows its type and knows its source as well as whether it is a row field,
```

Comment:
"source" is used without definition here, but maybe that's ok... not sure, maybe "its source matrix table or table"